ios/objc/src/TorchIValue.mm

```diff
@@ -0,0 +1,191 @@
+#include <vector>
+#import "TorchIValue.h"
+#import "TorchIValue+Internal.h"
+#import "TorchTensor.h"
+#import "TorchTensor+Internal.h"
+#import <Pytorch/Pytorch.h>
+
+#define IVALUE_TYPE(_) \
+    _(None) \
+    _(Bool) \
+    _(Int) \
+    _(Double) \
+    _(Tensor) \
+    _(BoolList) \
+    _(IntList) \
+    _(DoubleList)\
+    _(TensorList)
+
+#define IVALUE_SCALAR_TYPE(_) \
```

Comment:
The amount of code duplication isn't worth the macros. The code becomes less readable and tedious to debug and modify. Let's just copy-paste a few lines.